src/homework6/decor.py

```diff
@@ -0,0 +1,50 @@
+"""Параметрический декоратор.
+
+Создайте декоратор, который вызывает задекорированную функцию пока она не
+выполнится без исключений (но не более n раз - параметр декоратора). Если
+превышено количество попыток, должно быть возбуждено исключение типа
+TooManyErrors.
+"""
+import sys
+
+
+class TooManyErrors(Exception):
+    """Raised when function fails more than num times."""
+
+
+def too_many_errors(num):
+    """Parametric decorator.
+
+    :param num - number of allowed fails of decorated function.
+    Raises TooManyErrors exception and exits when num is reached.
+    """
+    def dec(func):
+        def wrapper(*args, **kwargs):
+            nonlocal num
+            try:
+                try:
+                    return func(*args, **kwargs)
+                except Exception:
+                    num -= 1
```

Comment:
in fact you can call function with this behavior only once

Reply:
next you'll have output like 
Function Works: 42/1=42.0
Function Works: 42/1=42.0
***Function FAILED, remaining attempts - -24039
Function Works: 42/2=21.0
Function Works: 42/2=21.0
***Function FAILED, remaining attempts - -24040

(I've deleted sys.exit from decorator)